src/components/graphs/UsageByMonthGraph.tsx

```diff
@@ -0,0 +1,204 @@
+import { useTheme } from '@mui/material';
+import {
+    eachMonthOfInterval,
+    isWithinInterval,
+    startOfMonth,
+    sub,
+} from 'date-fns';
+import { BarChart } from 'echarts/charts';
+import {
+    GridComponent,
+    LegendComponent,
+    MarkLineComponent,
+    TooltipComponent,
+} from 'echarts/components';
+import * as echarts from 'echarts/core';
+import { UniversalTransition } from 'echarts/features';
+import { CanvasRenderer } from 'echarts/renderers';
+import { useEffect, useMemo, useState } from 'react';
+import { useIntl } from 'react-intl';
+import {
+    useBilling_billingHistory,
+    useBilling_hydrated,
+} from 'stores/Billing/hooks';
+import useConstant from 'use-constant';
+import { CARD_AREA_HEIGHT, stripTimeFromDate } from 'utils/billing-utils';
+import useTooltipConfig from './useTooltipConfig';
+
+const chartContainerId = 'data-by-month';
+
+function UsageByMonthGraph() {
+    const theme = useTheme();
+    const intl = useIntl();
+    const tooltipConfig = useTooltipConfig();
+
+    const billingStoreHydrated = useBilling_hydrated();
+    const billingHistory = useBilling_billingHistory();
+
+    const [myChart, setMyChart] = useState<echarts.ECharts | null>(null);
+
+    const today = useConstant(() => new Date());
+
+    const months = useMemo(() => {
+        const startDate = sub(today, { months: 5 });
+
+        return eachMonthOfInterval({
+            start: startDate,
+            end: today,
+        }).map((date) => intl.formatDate(date, { month: 'short' }));
+    }, [intl, today]);
+
+    const seriesConfigs = useMemo(() => {
+        const startDate = startOfMonth(sub(today, { months: 5 }));
+
+        const filteredHistory = billingHistory.filter(({ billed_month }) => {
+            const billedMonth = stripTimeFromDate(billed_month);
+
+            return isWithinInterval(billedMonth, {
+                start: startDate,
+                end: today,
+            });
+        });
+
+        const data_series = filteredHistory.flatMap(
+            ({ billed_month, processed_data_gb }) => {
+                const billedMonth = stripTimeFromDate(billed_month);
+                const month = intl.formatDate(billedMonth, { month: 'short' });
+
+                return { month, data: processed_data_gb ?? 0 };
+            }
+        );
+
+        const hours_series = filteredHistory.flatMap(
+            ({ billed_month, task_usage_hours }) => {
+                const billedMonth = stripTimeFromDate(billed_month);
+                const month = intl.formatDate(billedMonth, { month: 'short' });
+
+                return { month, data: task_usage_hours ?? 0 };
+            }
+        );
+
+        return { data: data_series, hours: hours_series };
+    }, [billingHistory, intl, today]);
+
+    useEffect(() => {
```

Comment:
Please break this effect up into multiple pieces. `DataByHourGraph` is a good example.

At the very least we should keep the setting up `myChart` stuff separate from the option setting. This will allow us to more easily break this stuff up into reusable hooks in the future.